pkg/controllers/streaming/inmemorygateway_controller_test.go

```diff
@@ -336,16 +336,54 @@ func TestInMemoryGatewayReconciler(t *testing.T) {
 				StatusGatewayImage(testGatewayImage).
 				StatusProvisionerImage(testProvisionerImage),
 		},
+	}, {
+		Name: "conflicting gateway, owned",
+		Key:  testKey,
+		WithReactors: []rtesting.ReactionFunc{
+			rtesting.InduceFailure("create", "Gateway", rtesting.InduceFailureOpts{
+				Error: apierrs.NewAlreadyExists(schema.GroupResource{}, testName),
+			}),
+		},
+		GivenObjects: []rtesting.Factory{
+			inMemoryGatewayMinimal,
+			inMemoryGatewayImagesConfigMap,
+		},
+		APIGivenObjects: []rtesting.Factory{
+			gatewayGiven,
+		},
+		ShouldErr: true,
+		ExpectTracks: []rtesting.TrackRequest{
+			rtesting.NewTrackRequest(inMemoryGatewayImagesConfigMap, inMemoryGateway, scheme),
+		},
+		ExpectEvents: []rtesting.Event{
+			rtesting.NewEvent(inMemoryGateway, scheme, corev1.EventTypeWarning, "CreationFailed",
+				`Failed to create Gateway "%s":  "%s" already exists`, testName, testName),
```

Comment:
Why should the above event be emitted in this case?

If there is a good reason for emitting the above event, wouldn't it be helpful to give more information so that this case can be differentiated from the more serious problem of a non-owned child already existing?

Reply:
I'm a bit torn by this event. It distracts more than it informs, but it's also also accurate as to what the controller did. Excluding the event introduces a lot more complexity into the controller flow that, at least for now, is worth avoiding.